### ISSUE TITLE
Support for Server Side Rendering with Nuxt3

### ISSUE THREAD

**Issue by dalmia** [mengxiong10/vue-datepicker-next#25]
**Vue-datepicker-next version**: 1.0.2
**Vue version**: 3.2.37
**Browser**: Chrome

~**Reproduction Link or Source Code**~

Hey, thanks for creating this awesome library. I am trying to use this with Nuxt3 but it seems that it doesn't support SSR as the error mentions that it cannot find document. Am I correct in my understanding and if so, are there any plans to make it SSR friendly?

Thanks!

**Comment by papoola** [comment]
Create a plugin `plugins/date-picker.client.js`

```js
import DatePicker from 'vue-datepicker-next'
import 'vue-datepicker-next/index.css'

export default defineNuxtPlugin(nuxtApp => {
    nuxtApp.vueApp.component('DatePicker', DatePicker)
})
```

Then simply add this to your body

```html
<ClientOnly>
  <DatePicker v-model:value="myDate"></DatePicker>
</ClientOnly>
```

**Comment by DrPhil** [comment]
I couldn't get SSR with Vite to work. For some reason it then worked when changing the import to
```
import * as DatePicker from 'vue-datepicker-next/index.js';
```

Not sure if related. Hope this helps someone. 🤗

**Comment by vionijannet** [comment]
> Create a plugin `plugins/date-picker.client.js`
> 
> ```js
> import DatePicker from 'vue-datepicker-next'
> import 'vue-datepicker-next/index.css'
> 
> export default defineNuxtPlugin(nuxtApp => {
>     nuxtApp.vueApp.component('DatePicker', DatePicker)
> })
> ```
> 
> Then simply add this to your body
> 
> ```
> <ClientOnly>
>   <DatePicker v-model:value="myDate"></DatePicker>
> </ClientOnly>
> ```

This works for me, except the component shows later than any component and showing warning like this. 
```
[Vue warn]: Failed to resolve component: DatePicker
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
[Vue warn]: Component <Anonymous> is missing template or render function.
```
Still able to work though, but am I missing something here, or any idea to fix this?

**Comment by hiy2259** [comment]
use nuxt3, typescript 
### nuxt.config.ts
```typescript
export default defineNuxtConfig({
  devtools: { enabled: true },
  build: {
    transpile: ["vue-datepicker-next"],
  },
  plugins: [
    // your other plugins...
    { src: '~/plugins/vue-datepicker', mode: 'client' }, // client mode
  ],
})
```

### plugins/vue-datepicker.ts
```typescript
import DatePicker from 'vue-datepicker-next'
import 'vue-datepicker-next/index.css'

export default defineNuxtPlugin(nuxtApp => {
    nuxtApp.vueApp.component('DatePicker', DatePicker)
})
```

### datePickerTest.vue
```vue
<template>
    <div>
      <client-only><date-picker v-model:value="time0"></date-picker></client-only>
    </div>
  </template>

<script>
export default {
  data() {
    return {
      time0: null
    };
  },
};
</script>
```

**Comment by GalkinEvgeny** [comment]
> use nuxt3, typescript
> 
> ### nuxt.config.ts
> ```ts
> export default defineNuxtConfig({
>   devtools: { enabled: true },
>   build: {
>     transpile: ["vue-datepicker-next"],
>   },
>   plugins: [
>     // your other plugins...
>     { src: '~/plugins/vue-datepicker', mode: 'client' }, // client mode
>   ],
> })
> ```
> 
> ### plugins/vue-datepicker.ts
> ```ts
> import DatePicker from 'vue-datepicker-next'
> import 'vue-datepicker-next/index.css'
> 
> export default defineNuxtPlugin(nuxtApp => {
>     nuxtApp.vueApp.component('DatePicker', DatePicker)
> })
> ```
> 
> ### datePickerTest.vue
> ```
> <template>
>     <div>
>       <client-only><date-picker v-model:value="time0"></date-picker></client-only>
>     </div>
>   </template>
> 
> <script>
> export default {
>   data() {
>     return {
>       time0: null
>     };
>   },
> };
> </script>
> ```

Yeah, that's work fine if i use vue-datepicker-next directly, but what if it fails same way when other package (UI lib let's say) uses it? Is there a workaround to make other lib use this package client-only style but not make this lib fully client-only
